Main.py

```diff
@@ -379,13 +380,16 @@ def from_patch_file(settings, window=dummy_window()):
     output_path = os.path.join(output_dir, outfilebase)
 
     window.update_status('Patching ROM')
-    if extension == 'zpf':
-        subfile = None
+    if extension == 'patch':
+        apply_mbsdiff_patch_file(rom, settings.patch_file)
+        create_patch_file(rom, output_path + '.zpf')
     else:
-        subfile = 'P%d.zpf' % (settings.player_num)
-        if not settings.output_file:
-            output_path += 'P%d' % (settings.player_num)
-    apply_patch_file(rom, settings.patch_file, subfile)
+        subfile = None
+        if extension == 'zpfz':
+            subfile = f"P{settings.player_num}.zpf"
+            if not settings.output_file:
+                output_path += f"P{settings.player_num}.zpf"
```

Comment:
The original version didn't add any file extension here, so is that intended to be changed?

Reply:
Whoops, good catch.